sound/soc/intel/boards/sof_pcm512x.c

```diff
@@ -144,6 +144,36 @@ static int sof_pcm512x_codec_init(struct snd_soc_pcm_runtime *rtd)
 	return 0;
 }
 
+static int aif1_update_rate_den(struct snd_pcm_substream *substream,
+				struct snd_pcm_hw_params *params)
+{
+	struct snd_soc_pcm_runtime *rtd = substream->private_data;
+	struct sof_card_private *ctx = snd_soc_card_get_drvdata(rtd->card);
+	struct snd_ratnum *rats_no_pll;
+	unsigned int num = 0, den = 0;
+	int err;
+
+	rats_no_pll = kzalloc(sizeof(*rats_no_pll), GFP_KERNEL);
+	if (!rats_no_pll)
+		return -ENOMEM;
+
+	rats_no_pll->num = clk_get_rate(ctx->sclk) / 64;
+	rats_no_pll->den_min = 1;
+	rats_no_pll->den_max = 128;
+	rats_no_pll->den_step = 1;
```

Comment:
you initialise all 4 fields of `rats_no_pll` so you can just use `kmalloc()`. And in fact why does it have to be allocated? Wouldn't a local variable work?

Reply:
no reason, just existing code I took as is. Fixed with local structure.

Reply:
fixed